packages/nimble-components/src/rich-text-editor-viewer/specs/README.md

```diff
@@ -0,0 +1,377 @@
+
+The prototype includes the below functionalities,
+
+1. Basic formatting support.
+2. Image support. - The prototype includes [`Tiptap Extension`](https://tiptap.dev/api/nodes/image) to support images.
+3. `@mention` support. - The prototype includes default user list that gets triggered using `@` character.
+4. Markdown support. - The prototype includes support to render/retrieve content in markdown format.
+5. `Shadow root` support - The prototype uses [Microsoft Fast](https://github.com/microsoft/fast) to create the rich text editor as a
+   custom component that renders in the shadow root.
+6. `Top layer` support - Hyperlink using `nimble-dialog` which renders the popup in top layer.
+
+**_Note_**: This prototype is not made in a responsive fashion for smaller screens and is intended only for testing the specific functionalities listed
+above.
+
+### States
+
+_Button State_
+
+The buttons in the toolbar will indicate the formatting status of the selected text in the editor. For example, if the text in the editor
+is bold and the cursor is placed on the text, the bold button will be indicated as a active state. The active state is indicated as `block`
+appearance of the [`nimble-button`](https://nimble.ni.dev/storybook/?path=/docs/button--docs) and inactive state is in `ghost`
+appearance. Refer the below sample screenshot with active bold state.
+
+![Bold Button State](./spec-images/button-state.png)
+
+### Accessibility
+
+[Tiptap Accessibility](https://tiptap.dev/guide/accessibility)
+
+_Focus_
+
+-   Focus state of the editor will be the same as the `nimble-text-area`.
+-   Focus state of the buttons will be the same as the `nimble-button`.
+
+_Keyboard accessibility and shortcuts for text formatting_
+
+All the basic keyboard accessibility for the tiptap rich text editor is mentioned in their
+[keyboard shortcut](https://tiptap.dev/api/keyboard-shortcuts#predefined-keyboard-shortcuts) page. To facilitate easy access, listed down
+the supported text formatting keyboard shortcuts in the tiptap editor.
+
+| Key                     | Behavior                                                 |
+| ----------------------- | -------------------------------------------------------- |
+| Tab                     | To reach the editor                                      |
+| Shift + Arrow keys      | To select the group of texts in the editor               |
+| Ctrl + B                | To make the selected text bold                           |
+| Ctrl + I                | To make the selected text italic                         |
+| Ctrl + Shift + 7        | To enable the focused paragraph a numbered list          |
+| Ctrl + Shift + 8        | To enable the focused paragraph a bulleted list          |
+| Tab (For lists)         | To create a sub point within a bulleted or numbered list |
+| Shift + Tab (For lists) | To remove a sub point from a bulleted or numbered lists  |
+
+_Keyboard navigation with toolbar buttons focused_
```

Comment:
Switching between the buttons in the toolbar is missing here. For example, moving from the "bold" button to the "italic" button. Is that accomplished by pressing "Tab" or by pressing the "Right Arrow" key? I think it should be using the arrow keys, which means if you use the `nimble-toolbar` as I mentioned in another comment, you'll get that behavior for free.

Reply:
@mollykreis, We have tried using the `nimble-toolbar` in our prototype and here are some of the problems faced in the keyboard accessibility,
1. Consider having the focus in the editor text area and by pressing "Tab" will move the focus to icon buttons in the toolbar. Now by pressing "Arrow keys" to move the focus between formatting buttons. We also have `footer-actions` slot element in the same toolbar at the right end, which we specifically added for the comments feature ([mockup](https://www.figma.com/proto/Q5SU1OwrnD08keon3zObRX/SystemLink?type=design&node-id=6280-95369&scaling=min-zoom&page-id=2428%3A32954&starting-point-node-id=6280%3A94118&show-proto-sidebar=1)) to have `Cancel` and `Add comment` button. To reach the buttons in the slot, we can only use "Tab" because both the buttons are slotted into `slot = footer-actions` and the `footer-actions` is slotted into `slot = end` of nimble-toolbar. Is this making inconsistent in moving the focus by using "Arrow keys" for formatting options and "Tab" for slotted buttons which in the same toolbar?
2. From the comments feature point of view, even after we change the focus from formatting options (icon buttons) to `footer-actions` buttons by pressing "Tab", we couldn't move forward once after that. For an instance, consider the focus in the formatting options, pressing "Tab" moves the focus to `cancel` button and pressing "Tab" again, moves the focus back to the formatting options and not the `Add comment` button. 

Please let us know if there are any alternative design that we should consider to resolve these issue or is it fine to use the section or div tag to achieve the keyboard accessibility through "Tab" in all places.

Reply:
Thanks for the detailed information, we will use `nimble-toolbar` to wrap all the formatting options and wrap this toolbar and a slot element into another section to maintain it in a flexbox to stick with the mockups. I have updated the same in the [shadow DOM template](https://github.com/ni/nimble/blob/rich-text-editor-viewer-spec/packages/nimble-components/src/rich-text-editor/specs/README.md#anatomy) section of this spec. I have also updated the key accessibility which is to have "arrow keys" for the toolbar and "Tab" to switch to footer-action buttons.